Translators/WZDx/main.py

```diff
@@ -75,6 +75,66 @@ def translate(wzdx_geojson):
+        tim_body["highway"] = feature["properties"]["core_details"]["road_names"][0].replace("_", "-")
+        tim_body["id"] = feature["properties"]["core_details"]["name"].replace("_", "-")
+        tim_body["projectKey"] = 1
+        tim_body["buffers"] = []
+        active_tim_record = active_tim(feature, tim_body)
+        if active_tim_record:
+            logging.info(f"TIM already active for record: {tim_body['id']}")
+            continue
+        tims["timRwList"].append(tim_body)
+    return tims
+
+def active_tim(feature, tim_body):
+    tim_id = tim_body["id"]
+    # if TIM has an active TIM holding record that is current & info is the same as the current TIM record, then do not update
+    active_tim_holding = query_db(f"SELECT * FROM active_tim_holding WHERE client_id LIKE '%{tim_id}%'")
+    if len(active_tim_holding) > 0:
+        active_tim_holding = active_tim_holding[0]
+        return (active_tim_holding["direction"] == tim_body["direction"] and 
+            f"{active_tim_holding['start_latitude']:.8f}" == f"{tim_body['geometry'][0]['latitude']:.8f}" and 
+            f"{active_tim_holding['start_longitude']:.8f}" == f"{tim_body['geometry'][0]['longitude']:.8f}" and 
+            f"{active_tim_holding['end_latitude']:.8f}" == f"{tim_body['geometry'][-1]['latitude']:.8f}" and 
+            f"{active_tim_holding['end_longitude']:.8f}" == f"{tim_body['geometry'][-1]['longitude']:.8f}")
+
+    # if TIM has an active TIM record that is current & info is the same as the current TIM record, then do not update
+    active_tim = query_db(f"SELECT * FROM active_tim WHERE client_id LIKE '%{tim_id}%' AND tim_type_id = (SELECT tim_type_id FROM tim_type WHERE type = 'RW') AND marked_for_deletion = false")
+    if len(active_tim) > 0:
+        active_tim = active_tim[0]
+        return (active_tim["direction"] == tim_body["direction"] and
+            f"{active_tim['start_latitude']:.8f}" == f"{tim_body['geometry'][0]['latitude']:.8f}" and
+            f"{active_tim['start_longitude']:.8f}" == f"{tim_body['geometry'][0]['longitude']:.8f}" and
+            f"{active_tim['end_latitude']:.8f}" == f"{tim_body['geometry'][-1]['latitude']:.8f}" and
+            f"{active_tim['end_longitude']:.8f}" == f"{tim_body['geometry'][-1]['longitude']:.8f}")
+
+def delete_tims(feature_list):
+    # delete all active TIMs that are not in the current WZDx feed
+    active_tims = query_db("SELECT client_id FROM active_tim WHERE tim_type_id = (SELECT tim_type_id FROM tim_type WHERE type = 'RW') AND marked_for_deletion = false")
+    active_tims = [tim["client_id"] for tim in active_tims]
+    for tim in feature_list["timRwList"]:
+        if tim["id"] in active_tims:
+            active_tims.remove(tim["id"])
+    for tim in active_tims:
+        return_value = requests.delete(f'{os.getenv("ODE_ENDPOINT")}/rw-tim/{tim}', headers={"Accept": "application/json"})

```

Comment:
(blocking) Looks like a TIM Manager endpoint is being invoked using an address associated with the ODE. Will the address of the ODE & the TIM Manager always be the same?

Reply:
They will not - I've updated this env var to be correctly named as the translator will push to the TIM Manager instead of the ODE now

Reply:
Similar to the other PR, we'll want to update this to be the TIM Manager endpoint and not the ODE

Reply:
Updated